packages/ibm-products/src/components/Tearsheet/next/TearsheetBody.tsx

```diff
@@ -0,0 +1,191 @@
+/**
+ * Copyright IBM Corp. 2025
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import React, {
+  ForwardedRef,
+  forwardRef,
+  ReactNode,
+  useContext,
+  useRef,
+} from 'react';
+
+import { blockClass, TearsheetContext } from './context';
+import { SidePanel } from '../../SidePanel';
+import { Layer } from '@carbon/react';
+import { useCollapsible } from '../../../global/js/hooks/useCollapsible';
+/**
+ * ----------------
+ * TearsheetBody
+ * ----------------
+ */
+
+export interface TearsheetBodyProps {
+  /**
+   * Optional static content for body
+   */
+  children?: ReactNode;
+  /**
+   * Provide an optional class to be applied to the containing node.
+   */
+  className?: string;
+  /**
+   *You can provide content either through a callback (contentRenderer) or as static children—but not both.
+   * If both are provided, contentRenderer always takes precedence. This optional callback should return the content
+   * to be rendered in the body section, which can be either a single element or an array of elements based on your needs.
+   * If internal state access isn’t required, you can simply use static children instead
+   */
+}
+
+const TearsheetBody = forwardRef<HTMLDivElement, TearsheetBodyProps>(
+  ({ children, className }, ref: ForwardedRef<HTMLDivElement>) => {
+    return (
+      <div className={`${blockClass}__body ${className}`} ref={ref}>
+        {children}
+      </div>
+    );
+  }
+);
+
+export interface MainContentProps {
+  children: ReactNode;
+  className?: string;
+}
+/**
+ * ----------------
+ * MainContent
+ * ----------------
+ */
+export const MainContent = forwardRef<HTMLDivElement, MainContentProps>(
+  ({ children, className, ...rest }, ref: ForwardedRef<HTMLDivElement>) => {
+    const localRef = useRef<HTMLDivElement>(null);
+    const mainContentRef = ref || localRef;
+
+    const { setFullyCollapsed, disableHeaderCollapse } =
+      useContext(TearsheetContext);
+
+    const container =
+      typeof mainContentRef === 'function'
+        ? null
+        : (mainContentRef?.current ?? null);
+
+    const collapseHeader = (collapse: boolean) => {
+      if (container) {
+        if (collapse) {
+          const canScroll = container.scrollHeight > container.clientHeight; // collapse header only when there is scroll
+
+          if (canScroll) {
+            setFullyCollapsed?.(true);
+          }
+        } else if (container.scrollTop === 0) {
+          // expand header when scroll reaches top
+          setFullyCollapsed?.(false);
+        }
+      }
+    };
+
+    useCollapsible({
```

Comment:
The collapse animation still feels like it jumps a little bit, lets make sure to remove with @RichKummer